diff --git a/gtest/include/gtest/internal/gtest-port.h b/gtest/include/gtest/internal/gtest-port.h
--- a/gtest/include/gtest/internal/gtest-port.h
+++ b/gtest/include/gtest/internal/gtest-port.h
@@ -253,6 +253,9 @@
 # if defined __ANDROID__
 #  define GTEST_OS_LINUX_ANDROID 1
 # endif
+#elif defined __FreeBSD__
+# define GTEST_OS_LINUX 1
+# define GTEST_HAS_CLONE 0
 #elif defined __MVS__
 # define GTEST_OS_ZOS 1
 #elif defined(__sun) && defined(__SVR4)
diff --git a/gtest/include/gtest/internal/gtest-internal.h b/gtest/include/gtest/internal/gtest-internal.h
--- a/gtest/include/gtest/internal/gtest-internal.h
+++ b/gtest/include/gtest/internal/gtest-internal.h
@@ -44,7 +44,11 @@
 # include <sys/types.h>
 # include <sys/wait.h>
 # include <unistd.h>
+# include <signal.h>
 #endif  // GTEST_OS_LINUX
+#if GTEST_CAN_STREAM_RESULTS_
+# include <sys/socket.h>
+#endif  // GTEST_CAN_STREAM_RESULTS_
 
 #include <ctype.h>
 #include <string.h>